expression/builtin_cast_vec.go

```diff
@@ -388,11 +388,44 @@ func (b *builtinCastIntAsStringSig) vecEvalString(input *chunk.Chunk, result *ch
 }
 
 func (b *builtinCastRealAsIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastRealAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	f64s := buf.Float64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if !mysql.HasUnsignedFlag(b.tp.Flag) {
```

Comment:
How about using a local variable to store this result outside this loop so that we don't need to call this function each time.

Reply:
@tangwz 
I think that you misunderstand what @qw4990 said, we could cache the result of `mysql.HasUnsignedFlag(b.tp.Flag)`, here's a demo.

```go
unsigned := mysql.HasUnsignedFlag(b.tp.Flag)
for {
    if !unsigned {
        ...
    }
}
```

Reply:
my bad. I will fix it.